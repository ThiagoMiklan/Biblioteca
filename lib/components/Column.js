"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: "string"
};

var Column = function Column(props) {
  (0, _type_validations.validate)(props, props_obj, "Column");
  var definition = props.definition == undefined ? "column" : "column " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, props.children);
};

var _default = Column;
exports.default = _default;