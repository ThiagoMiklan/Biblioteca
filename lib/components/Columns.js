"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: 'string'
};

var Columns = function Columns(props) {
  (0, _type_validations.validate)(props, props_obj, "Columns");
  var definition = props.definition == undefined ? "columns" : "columns " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, React.Children.map(props.children, function (child, i) {
    return child;
  }));
};

var _default = Columns;
exports["default"] = _default;