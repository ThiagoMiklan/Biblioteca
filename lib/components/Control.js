"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: 'string'
};

var Control = function Control(props) {
  (0, _type_validations.validate)(props, props_obj, "Control");
  var definition = props.definition == undefined || props.definition == '' ? "control" : "control " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, React.Children.map(props.children, function (child, i) {
    return child;
  }));
};

var _default = Control;
exports.default = _default;