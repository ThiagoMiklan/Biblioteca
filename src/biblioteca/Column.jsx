import React from 'react';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import PropTypes from 'prop-types';
/**
 * Componente inspirado na classe column do bulma
 * Utilizado como filho do componente Columns
 * Possui filhos, pois funciona como um container para outros
 * elementos
 * Sem eventos
 */

const Column = (props) => {
    var definition = classnames(getClassName("column "+ props.definition, "Column"));
    return (
        <div className={definition}>{props.children}</div>
    );
}

Column.propTypes = {
    // caracter√≠sticas do bulma para a coluna
    definition : PropTypes.string
}

export default Column;