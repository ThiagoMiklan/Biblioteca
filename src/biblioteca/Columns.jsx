import React from 'react';
import getClassName from '../tools/getClassName';
import classnames from 'classnames';
import PropTypes from 'prop-types';

/**
 *  Funciona como um container para Column
 *  Baseado na classe Columns do Bulma
 *  Sem eventos
 */

const Columns = (props)=>{
    var definition = classnames(getClassName("columns "+ props.definition,"Columns"));

    return <div className={definition}>
        {
            React.Children.map(props.children, (child, i)=>{
                return child;
            })
        }
    </div>

}

Columns.propTypes = {
    // Caracter√≠sticas do Bulma
    definition: PropTypes.string
}



export default Columns