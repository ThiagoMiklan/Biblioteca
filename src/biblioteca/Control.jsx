// @flow
import * as React from 'react'
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import checkerDefinition from '../tools/checker.js';

/*
    - Componente que funciona como container
    - Sem eventos
*/

type Props ={
    definition?:string,
    children?: React.Node
}

const Control = (props:Props)=>{
    checkerDefinition(props.definition,"Control");
    
    var definition = getDefinition(props.definition);
    return(
        <div className={definition}>
              {
                    React.Children.map(props.children, (child, i) => {
                        return child
                    })
                }
        </div>
    );
}

function getDefinition(definition){
    if(definition == undefined || definition == ''){
        return classnames(getClassName("control icons-left","Control"));
    }else{
        return classnames(getClassName("control "+ definition,"Control"));
    }
}


Control.propTypes = {
    // Caracter√≠sticas do Bulma 
    definition: PropTypes.string
}

export default Control;
