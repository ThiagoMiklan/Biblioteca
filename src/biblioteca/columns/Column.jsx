import React from 'react';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName';
import PropTypes from 'prop-types';

const Column = (props) => {
    var definition = classnames(getClassName("column "+ props.definition, "Column"));
    return (
        <div className={definition}>{props.children}</div>
    );
}

Column.propTypes = {
    definition : PropTypes.string
}

export default Column;