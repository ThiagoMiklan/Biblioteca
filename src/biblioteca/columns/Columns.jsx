import React from 'react';
import getClassName from '../../tools/getClassName';
import classnames from 'classnames';


const Columns = (props)=>{
    var definition = classnames(getClassName("columns "+ props.definition,"Columns"));

    return <div className={definition}>
        {
            React.Children.map(props.children, (child, i)=>{
                return child;
            })
        }
    </div>

}

export default Columns