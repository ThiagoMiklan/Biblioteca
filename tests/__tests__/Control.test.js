import React from 'react';
import renderer from 'react-test-renderer';
import Control from '../../src/components/Control';
import Columns from '../../src/components/Columns';
import Column from '../../src/components/Column';

const tags = [
  {value:'One', definition:'is-primary'},
  {value:'Two', definition:'is-success'},
  {value:"Three",definition:'is-info'},
  {value:"Four",definition:'is-link'},
  {value:"Five",definition:'is-warning'},
  {value:"Six", definition:'vdanger'}
];
const tags_addons =[
  {value:'Package'},
  {value:'Bulma', definition:'is-primary'},
]


test('control normal',()=>{
    const component = renderer.create(
         <div className="container is-fluid">
          <Columns>
          <Column definition="is-half is-centered">
          <div class="field">
            <label class="label">Name</label>
            <Control>
              <input class="input" type="text" placeholder="Text input"/>
          </Control>
          </div>

          <div class="field">
            <label class="label">Subject</label>
            <Control>
              <div class="select">
                <select>
                  <option>Select dropdown</option>
                  <option>With options</option>
                </select>
              </div>
            </Control>
          </div>

          <div class="field">
            <Control> 
              <label class="radio">
              <input type="radio" name="question"/>
                Yes
              </label>
              <label class="radio">
              <input type="radio" name="question"/>
                 No
              </label>
            </Control>
          </div>
          </Column>
          </Columns>
         </div>     
  ).toJSON();
    expect(component).toMatchSnapshot();
});
