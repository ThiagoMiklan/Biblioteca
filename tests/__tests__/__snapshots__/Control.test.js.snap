// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`control normal 1`] = `
<div
  className="container is-fluid"
>
  <div
    className="columns"
  >
    <div
      className="column is-half is-centered"
    >
      <div
        class="field"
      >
        <label
          class="label"
        >
          Name
        </label>
        <div
          className="control"
        >
          <input
            class="input"
            placeholder="Text input"
            type="text"
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          class="label"
        >
          Subject
        </label>
        <div
          className="control"
        >
          <div
            class="select"
          >
            <select>
              <option>
                Select dropdown
              </option>
              <option>
                With options
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <div
          className="control"
        >
          <label
            class="radio"
          >
            <input
              name="question"
              type="radio"
            />
            Yes
          </label>
          <label
            class="radio"
          >
            <input
              name="question"
              type="radio"
            />
            No
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
